pkg/framework/machinesets.go

```diff
@@ -312,6 +313,16 @@ func WaitForMachineSet(c client.Client, name string) {
 				name, len(machines), int(replicas))
 		}
 
+		failed := FilterMachines(machines, MachinePhaseFailed)
+		if len(failed) > 0 {
+			// if there are failed machines, print them out and exit the test
+			msg := fmt.Sprintf("found %d Machines in failed phase: ", len(failed))
+			for _, m := range failed {
+				msg = fmt.Sprintf("%s %s", msg, m.Name)
+			}
```

Comment:
I wonder if we can dump the Machine status out here as well, it would be good to see why the Machine failed. Don't we normally put an error message on the status when the machine fails?

Reply:
that's a good suggestion, i'll poke around to see if it can be more consistent with other failure messages.

Reply:
i looked through some must-gathers and it looks like we do put something in the `status.conditions`, i wonder if we should just print all the available `message` fields from those for each machine?

Reply:
i'm doing a little more digging in MAO to see if i can come up with a strategy for this.